server/src/main/java/org/elasticsearch/search/SearchService.java

```diff
@@ -316,6 +327,12 @@ public SearchService(
         defaultSearchTimeout = DEFAULT_SEARCH_TIMEOUT_SETTING.get(settings);
         clusterService.getClusterSettings().addSettingsUpdateConsumer(DEFAULT_SEARCH_TIMEOUT_SETTING, this::setDefaultSearchTimeout);
 
+        minimumDocsPerSlice = MINIMUM_DOCS_PER_SLICE.get(settings);
+        if (MINIMUM_DOCS_PER_SLICE == clusterService.getClusterSettings().get(MINIMUM_DOCS_PER_SLICE.getKey())) {
+            // Only add it if the setting has been register which should only happen on tests
+            clusterService.getClusterSettings().addSettingsUpdateConsumer(MINIMUM_DOCS_PER_SLICE, this::setMinimumDocsPerSlice);
```

Comment:
I did it this way because I wasn't sure how to set a static setting from inside the test and with a dynamic setting is straight forward. But yes, if it easy, it can be changed.

Reply:
got it, I would think that you need to add it to the gradle configuration that creates the cluster.

Reply:
for ESSingleNodeTestCase, settings are defined in newNode